independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/ConfigService.java

```diff
@@ -0,0 +1,194 @@
+package io.quarkus.analytics;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+
+import io.quarkus.analytics.config.FileLocations;
+import io.quarkus.analytics.config.GroupIdFilter;
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.LocalConfig;
+import io.quarkus.analytics.dto.config.NoopRemoteConfig;
+import io.quarkus.analytics.dto.config.RemoteConfig;
+import io.quarkus.analytics.rest.ConfigClient;
+import io.quarkus.analytics.util.FileUtils;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+/**
+ * Decided the build analytics behaviour. Retrieves, stores and provides the configuration.
+ */
+public class ConfigService {
+    public static final String QUARKUS_ANALYTICS_DISABLED_LOCAL_PROP = "quarkus.analytics.disabled";
+    public static final String QUARKUS_ANALYTICS_PROMPT_TIMEOUT = "quarkus.analytics.prompt.timeout";
+    private static final int DEFAULT_REFRESH_HOURS = 12;
+    private static final String NEW_LINE = System.lineSeparator();
+    static final String ACCEPTANCE_PROMPT = NEW_LINE +
+            "*****************************************************************************************************" + NEW_LINE
+            + " Do you agree to send anonymous usage data to the Quarkus community to help improve Quarkus? (y/n) " + NEW_LINE;
```

Comment:
Should we link here to the data collection notice, explaining what is gathered?

Reply:
I was thinking we make a quarkus.io/usage landing page with such details. wdyt?

should start that PR too.

Reply:
Here's the internal document: https://docs.google.com/document/d/1LzjmxL-8cnybpcg8vNCu-mNmu9y5qbhaM3fWpZh97mg/edit. I assume we need to copy that verbatim to a web page? Right @maxandersen?

Reply:
I think we should place a link to the text. It can be a gist on github.

Reply:
How about we make a regular web page on quarkus.io/usage with the content of https://docs.google.com/document/d/1LzjmxL-8cnybpcg8vNCu-mNmu9y5qbhaM3fWpZh97mg/edit? Then we link to that from the prompt?

Reply:
Sounds good to me. Will create a PR for it.

Reply:
https://github.com/quarkusio/quarkusio.github.io/pull/1721